### DISCUSSION TITLE
Reset player size via command after walking through a portal

### DISCUSSION THREAD

**Discussion by dummoprogrammer** [iPortalTeam/ImmersivePortalsMod#1661] [Q&A]
So, I'm in the middle of making a parkour map utilizing the mace, and I decided to whip out Immersive Portals for some cool visual effects. But I ran into a problem with the player scaling, because I want to reset the scaling once the player has walked through the portal, probably via use of commands.

I've been messing around with Immersive Portals since early 2022, and I've always relied on the Pehkui mod managing the scaling that happens when you walk through portals with different sizes. But this seems like the first version (that I've noticed) that doesn't use Pehkui anymore. I literally removed the mod from my folder and the game still function identically to how it did before.

I did some "testing", and it seems that iPortal is using the vanilla `minecraft:generic.scale` attribute to change the player's size on the fly. When running through a portal reducing my size by a half, and running `/attribute @s minecraft:generic.scale get`, it returns 0.5. But when I query the base value instead by running `/attribute @s minecraft:generic.scale base get`, it always returns 1.0, and it seems that Minecraft only allows me to change the base scale.

I don't know if iPortal is using some kind of modifier on the attribute or if it's writing to it directly from the source code, but I need some way to reset the player size by use of commands. I know I can just modify the base value to counteract the portal scaling, but I will be dealing with variable width portals throughout the entire map (the first one is a 5/19 scale already), so that's out of the question.

Bottom line (literally and figuratively), does anyone know how to reset the player scale using commands?

**Comment by qouteall** [comment]
Yes iPortal use a scale modifier `iportal:scaling`

You can remove scaling by `/attribute @s minecraft:generic.scale modifier remove iportal:scaling`

https://qouteall.fun/immptl/wiki/Spatial-Transformation.html#teleportation-1
